src/poliastro/core/spheroid_location.py

```diff
@@ -177,11 +177,13 @@ def cartesian_to_ellipsoidal(_a, _c, x, y, z):
     e2_ = e2 / (1 - e2)
     p = np.sqrt(x ** 2 + y ** 2)
     th = np.arctan(z * _a / (p * _c))
-    lon = np.arctan(y / x)
+    lon = np.arctan2(y, x)  # Use `arctan2` so that lon lies in the range: [-pi, +pi]
     lat = np.arctan((z + e2_ * _c * np.sin(th) ** 3) / (p - e2 * _a * np.cos(th) ** 3))
 
     v = _a / np.sqrt(1 - e2 * np.sin(lat) ** 2)
-    h = p / np.cos(lat) - v
-    h = z / np.sin(lat) - (1 - e2) * v
+    if lat == np.pi / 2 or lat:
+        h = z / np.sin(lat) - (1 - e2) * v
+    else:
+        h = x / np.cos(lat) - v
```

Comment:
Where does this formula come from?

Reply:
I saw two equivalent representations in the original Bowring paper:

![image](https://user-images.githubusercontent.com/68844397/126833601-0858b000-7d92-47cc-a33b-3e2132cf49c2.png)